clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -568,6 +638,31 @@ public void shouldPreserveCommittedConsumerGroupsOffsetsOnAbortIfTransactionsAre
         assertThat(producer.consumerGroupOffsetsHistory(), equalTo(Collections.singletonList(expectedResult)));
     }
 
+    @Test
+    public void shouldPreserveOffsetsFromCommitByGroupMetadataOnAbortIfTransactionsAreEnabled() {
+        buildMockProducer(true);
+        producer.initTransactions();
+        producer.beginTransaction();
+
+        String group = "g";
+        Map<TopicPartition, OffsetAndMetadata> groupCommit = new HashMap<TopicPartition, OffsetAndMetadata>() {
+            {
+                put(new TopicPartition(topic, 0), new OffsetAndMetadata(42L, null));
+                put(new TopicPartition(topic, 1), new OffsetAndMetadata(73L, null));
+            }
+        };
+        producer.sendOffsetsToTransaction(groupCommit, groupMetadata(group));
+        producer.commitTransaction();
+
+        producer.beginTransaction();
+        producer.abortTransaction();
```

Comment:
Seems this test case would be more interesting if we tried to commit a separate set of offsets before aborting

Reply:
ack